### DISCUSSION TITLE
How to install TimescaleDB Comunity Edition with Supabase's Postgres image?

### DISCUSSION THREAD

**Discussion by haleksandre** [supabase/supabase#30086] [Questions]
Has anyone figured out a simple way to integrate TimescaleDB Community Edition inside Supabase's Postgres image or a multi-stage build image?

I've tried a few things but always encounter errors at different stages.

I initially tried a multi-stage build using both Supabase's Postgres image and TimescaleDB's image, but can't seem to copy the files to the right places since Supabase's image uses Nix. Same issue when installing directly using `apt-get`. Then, I tried to leverage Nix by either directly installing or using a flake.nix (`RUN nix profile install [...]`) to install the packages inside the image, although Postgres sees the new version of TimescaleDB (`SELECT * FROM pg_available_extensions;`), I can't seem able to enable it. Running the SQL statement `CREATE extension timescaledb;` gives me an error `file not found`.

Does anyone know how to do it or can point me in the right direction, it would be greatly appreciated.

**Comment by paulwer** [comment]
have you found a solution?

**Reply by haleksandre** [reply]
I resorted to building my own image with a multi-stage build image, installing/building from source the various extensions manually within the Postgres/Rust image, pulling the Supabase migration files from their own Postgres image, and copying over those plugins files within Postgres default directory, as well as copying those migration files within the default docker-entrypoint, with an extra migration file that enables the plugin, directory within the final stage using TimeScale's image.

**Reply by paulwer** [reply]
are you able to share it?

**Reply by haleksandre** [reply]
The image was originally built for working with our own databases and add-ons, but I put together a stripped-down version here: https://github.com/haleksandre/postgres.

I used to build it with GitLab CI and host it on GitLab.com, so I haven’t tested if this exact version works as-is. But it should give you a starting point for building your own multi-stage Containerfile.
